File: backend/beforeselling/src/main/java/kakaotech/kangwon3/beforeselling/domains/regulation/domain/entity/MedicalDevice.java
package kakaotech.kangwon3.beforeselling.domains.regulation.domain.entity;

import java.time.OffsetDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.Builder;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "medical_devices")
public class MedicalDevice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "medical_device_id")
    private Long id;

    @Column(name = "device_sn", length = 30, nullable = false, unique = true)
    private String deviceSn;

    @Column(name = "product_name", length = 300, nullable = false)
    private String productName;

    @Column(name = "classification_no", length = 30)
    private String classificationNo;

    @Column(name = "grade", length = 5)
    private String grade;

    @Column(name = "permission_type", length = 20)
    private String permissionType;

    @Column(name = "industry", length = 30)
    private String industry;

    @Column(name = "purpose", columnDefinition = "TEXT")
    private String purpose;

    @Column(name = "fetched_at", nullable = false)
    private OffsetDateTime fetchedAt;

    @Builder(access = AccessLevel.PRIVATE)
    private MedicalDevice(String deviceSn,
                          String productName,
                          String classificationNo,
                          String grade,
                          String permissionType,
                          String industry,
                          String purpose,
                          OffsetDateTime fetchedAt) {
        this.deviceSn = deviceSn;
        this.productName = productName;
        this.classificationNo = classificationNo;
        this.grade = grade;
        this.permissionType = permissionType;
        this.industry = industry;
        this.purpose = purpose;
        this.fetchedAt = fetchedAt;
    }

    public static MedicalDevice fromApi(String deviceSn,
                                        String productName,
                                        String classificationNo,
                                        String grade,
                                        String permissionType,
                                        String industry,
                                        String purpose) {
        return MedicalDevice.builder()
                .deviceSn(deviceSn)
                .productName(productName)
                .classificationNo(classificationNo)
                .grade(grade)
                .permissionType(permissionType)
                .industry(industry)
                .purpose(purpose)
                .fetchedAt(OffsetDateTime.now())
                .build();
    }
}

File: ai/app/models/medical_device.py
from sqlalchemy import Index, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class MedicalDevice(Base):
    """식약처 의료기기 품목정보."""

    __tablename__ = "medical_devices"
    __table_args__ = (
        Index("ix_medical_devices_classification_no", "classification_no"),
        Index("ix_medical_devices_product_name", "product_name"),
    )

    medical_device_id: Mapped[int] = mapped_column(
        primary_key=True, autoincrement=True,
    )
    device_sn: Mapped[str] = mapped_column(String(30), unique=True, nullable=False)
    product_name: Mapped[str] = mapped_column(String(300), nullable=False)
    classification_no: Mapped[str | None] = mapped_column(String(30))
    grade: Mapped[str | None] = mapped_column(String(5))
    permission_type: Mapped[str | None] = mapped_column(String(20))
    industry: Mapped[str | None] = mapped_column(String(30))
    purpose: Mapped[str | None] = mapped_column(Text)
